DataWorkerService/Processors/Resolvers/Implementations/TournamentProcessorResolver.cs

```diff
@@ -10,17 +11,29 @@ IEnumerable<IProcessor<Tournament>> processors
 {
     public IProcessor<Tournament> GetDataProcessor() =>
         Processors.FirstOrDefault(p => p is TournamentDataProcessor)
-            ?? throw new InvalidOperationException($"Processor was not registered: {nameof(TournamentDataProcessor)}");
+        ?? throw new InvalidOperationException($"Processor was not registered: {nameof(TournamentDataProcessor)}");
 
     public override IProcessor<Tournament> GetAutomationChecksProcessor() =>
         Processors.FirstOrDefault(p => p is TournamentAutomationChecksProcessor)
-        ?? throw new InvalidOperationException($"Processor was not registered: {nameof(TournamentAutomationChecksProcessor)}");
+        ?? throw new InvalidOperationException(
+            $"Processor was not registered: {nameof(TournamentAutomationChecksProcessor)}");
 
     public IProcessor<Tournament> GetStatsProcessor() =>
         Processors.FirstOrDefault(p => p is TournamentStatsProcessor)
         ?? throw new InvalidOperationException($"Processor was not registered: {nameof(TournamentStatsProcessor)}");
 
+    public IProcessor<Tournament> GetNextProcessor(TournamentProcessingStatus processingStatus) =>
+        processingStatus switch
+        {
+            TournamentProcessingStatus.NeedsMatchData => GetDataProcessor(),
+            TournamentProcessingStatus.NeedsAutomationChecks => GetAutomationChecksProcessor(),
+            TournamentProcessingStatus.NeedsStatCalculation => GetStatsProcessor(),
+            TournamentProcessingStatus.NeedsVerification => GetVerificationProcessor(),
+            _ => throw new Exception("No processor")
```

Comment:
Error message should be more descriptive. Maybe use `InvalidArgumentException` as well?